modules/Bio/EnsEMBL/Compara/RunnableDB/SpeciesTree/CheckSketches.pm

```diff
@@ -155,7 +160,7 @@ sub find_file_for_gdb {
 	
 	$suffixes = [''] unless defined $suffixes->[0];
 
-	my $prefix = $gdb->name . "." . $gdb->assembly;
+	my $prefix = basename($gdb->_get_genome_dump_path($self->param('genome_dumps_dir')));
 	foreach my $suffix ( @$suffixes ) {
 		my @found_files = glob "$dir/$prefix.*$suffix";
```

Comment:
The use of `_get_genome_dump_path` here was to get the filename, which the `.msh` files mimic - there is no api call for generating the filename separate from the path. The reason for not using the method itself to generate the whole path was concern over this line: `my $subdir = $self->dbID ? Bio::EnsEMBL::Hive::Utils::dir_revhash($self->dbID) : '';`. To me it appears that it will add a subdir if the genome_db has an ID? The `.msh` files so far as I can tell are never in subdirs?

Reply:
:+1: forgot `dir_rev_hash` - although not in this scope, I wonder if this is something we should consider adding as an optional flag for $subdir.